scripts/integration-script.sh

```diff
@@ -37,28 +56,59 @@ clone_repo() {
 
 # Loop through each repository and clone it
 for repo_info in "${repos[@]}"; do
-    # Split the repo_info string into url, branch, target_dir, and optional destination
-    IFS=' ' read -r -a repo <<< "$repo_info"
-    clone_repo "${repo[0]}" "${repo[1]}" "${repo[2]}"
-    echo -e "Cloned ${repo[0]}@${repo[1]} into ${repo[2]}\n"
-
-    # Copy the repo's content:
-    # - default: from "docs"  -> $TARGET_DIR
-    # - if repo[3] is set (e.g., "tutorials"): from "docs/<dest>" -> $TARGET_DIR/<dest>
-    dest="${repo[3]:-}"
-    src_subpath="docs${dest:+/$dest}"
-    dest_dir="$TARGET_DIR${dest:+/$dest}"
-
+  IFS=' ' read -r -a repo <<< "$repo_info"
+  repo_url="${repo[0]}"
+  branch="${repo[1]}"
+  target_dir="${repo[2]}"
+  dest="${repo[3]:-}" # optional, e.g. 'tutorials'
+
+  key="$(repo_key_from_url "$repo_url")"
+  # Fallback if URL produced an empty key (e.g., trailing slash edge cases)
+  if [[ -z "$key" ]]; then
+    key="${target_dir##*/}"
+  fi
+  src_subpath="docs${dest:+/$dest}"
+  dest_dir="$TARGET_DIR${dest:+/$dest}"
+  # If this is the tutorials destination, ensure a clean slate once
+  if [[ -n "$dest" && "$dest" == "tutorials" && -z "${CLEANED_TUTORIALS:-}" ]]; then
+    rm -rf "$dest_dir"
     mkdir -p "$dest_dir"
-    if [[ -d "${repo[2]}/$src_subpath" ]]; then
-      cp -r "${repo[2]}/$src_subpath/"* "$dest_dir/"
+    CLEANED_TUTORIALS=1
+  fi
```

Comment:
This removes the tutorial content that is in the uds-docs repo itself I think? Checking the preview site most of the tutorials 404 (https://deploy-preview-321--uds.netlify.app/tutorials/deploy-with-uds-core/) with just the RKE2 one left (the only external one).